addons/web/static/src/core/dialog/dialog.js

```diff
@@ -3,8 +3,38 @@
 import { useHotkey } from "@web/core/hotkeys/hotkey_hook";
 import { useActiveElement } from "../ui/ui_service";
 import { useForwardRefToParent } from "@web/core/utils/hooks";
+import { Component, useChildSubEnv, useExternalListener, useState } from "@odoo/owl";
+import { throttleForAnimation } from "@web/core/utils/timing";
+import { makeDraggableHook } from "../utils/draggable_hook_builder";
+
+const useDialogDraggable = makeDraggableHook({
+    name: "useDialogDraggable",
+    onDragStart({ ctx, getRect }) {
+        const { top, left } = getRect(ctx.current.element);
+        ctx.current.startPosition = { top, left };
```

Comment:
I can't use the coordinates of the dialog rect because those coordinates are relative to the viewport not like the position of the dialog (`.modal-content`) which is relative to its parent `.modal-dialog` and its `margin: 0 auto;`.
This makes `viewport.origin !== dialog.origin`